tests/e2escenarios/e2e_test.go

```diff
@@ -270,4 +294,97 @@ var _ = Describe("E2E Test", func() {
 		})
 	})
+
+	Context("starting with non-empty Directory add Binding", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			commonVar.CliRunner.EnsureOperatorIsInstalled("service-binding-operator")
+			commonVar.CliRunner.EnsureOperatorIsInstalled("cloud-native-postgresql")
+			Eventually(func() string {
+				out, _ := commonVar.CliRunner.GetBindableKinds()
+				return out
+			}, 120, 3).Should(ContainSubstring("Cluster"))
+			helper.Chdir(commonVar.Context)
+			helper.CopyExample(filepath.Join("source", "devfiles", "go"), commonVar.Context)
+			addBindableKind := commonVar.CliRunner.Run("apply", "-f", helper.GetExamplePath("source", "devfiles", "go", "cluster.yaml"))
+			Expect(addBindableKind.ExitCode()).To(BeEquivalentTo(0))
+		})
+
+		It("should verify developer workflow of using binding as env in innerloop", func() {
+			bindingName := helper.RandString(6)
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				// helper.ExpectString(ctx, "Based on the files in the current directory odo detected")
+				helper.ExpectString(ctx, "Language: Go")
+				helper.ExpectString(ctx, "Project type: Go")
+				helper.ExpectString(ctx, "Is this correct")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+
+			//add binding information (binding as file)
+			helper.Cmd("odo", "add", "binding", "--name", bindingName, "--service", "cluster-example-initdb", "--bind-as-files=false").ShouldPass()
+
+			// // "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			// "send data"
+			data := sendDataEntry(ports["8080"])
+			Expect(data["message"]).To(Equal("User created successfully"))
+
+			// "get all data"
+			rec := receiveData(fmt.Sprintf(ports["8080"] + "/api/user"))
+			helper.MatchAllInOutput(rec, []string{"id", "1", "name", "joe", "location", "tokyo", "age", "23"})
+
+			// check odo describe to check for env
+			stdout := helper.Cmd("odo", "describe", "binding").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{"Available binding information:", "CLUSTER_HOST", "CLUSTER_PASSWORD", "CLUSTER_USERNAME"})
+
+			// "running odo list"
+			stdout = helper.Cmd("odo", "list").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{componentName, "Go", "Dev", bindingName})
+
+			// "exit dev mode and run odo deploy"
```

Comment:
Can you update this comment? Because we are not running `odo deploy` after ;-)
Also, can you add a TODO comment here, as a reminder that we needed to stop the Dev Session before removing the binding because of #6101?

Reply:
> Also, can you add a TODO comment here, as a reminder that we needed to stop the Dev Session before removing the binding because of https://github.com/redhat-developer/odo/issues/6101?

We are stopping the odo dev process before removing binding, do you want me to add the issue about the adding the link to the https://github.com/redhat-developer/odo/issues/6101?

Reply:
Yes, that would be great as a reminder. Thanks.

Reply:
Updated